### ISSUE TITLE
Duplicate prompts when equipping/unequpping some items

### ISSUE THREAD

**Issue by nlavsky** [crawl/crawl#2916]
0.30-a0-605-gb2882149bf

After the introduction of unified equip/unequip menus, the game repeats the same prompt if you try to replace one dangerous item with another one.

Switching dangerous amulets via the `P` menu shows three prompts instead of two:

```
Really put on r - the amulet of Vaghepna {Contam Fly Regen+ rCorr Stlth+}?
Really remove l - the amulet "Foemaes" {Reflect Drain rPois rN+ Str-3 Int+5 SH+5}?
Really remove l - the amulet "Foemaes" {Reflect Drain rPois rN+ Str-3 Int+5 SH+5}?
```

If you try to switch items inscribed with `!*` via the `e` menu, you'll get four prompts instead of two:

```
Really equip p - an amulet of regeneration {!*}?
Really put on p - an amulet of regeneration {!*}?
Really remove h - an amulet of reflection {!*}?
Really remove h - an amulet of reflection {!*}?
```

Similarly, removing an amulet of Faith prompts twice:
* ``Really remove ... Really remove ...``, when removing it via the `R` menu
* or ``Really choose ... Really remove ...``, when using the `c` menu.

The bug happens for weapons, armour, and jewellery.

**Comment by nlavsky** [comment]
After a4aca97477, there is no duplicate prompt when removing an amulet of Faith via `R` command.  But you get two prompts when removing Faith via `c` still:

```
Really unequip F - an amulet of faith?
Really remove F - an amulet of faith?
```

Swapping an amulet with Drain with another amulet with Drain still prompts three times instead of two (both via `e` and via `P`):

```
Really put on U - the amulet "Qish" {Harm *Corrode Drain Regen+ Int+10}?
Really remove r - the amulet of the Delightful Apricot {Drain Will+ Regen+ rCorr Stlth+}?
Really remove r - the amulet of the Delightful Apricot {Drain Will+ Regen+ rCorr Stlth+}?
```

For weapons and armour, the bug happens only when you have items inscribed with `!*` and only when swapping them via `e` (swapping via `w` or `W` is fine):

```
Really equip b - a +0 dagger {!*}?
Really wield b - a +0 dagger {!*}?
Really unwield a - a +0 short sword {!*}?

Really equip i - a +0 robe {!*}?
Really wear i - a +0 robe {!*}?
Really take off d - a +0 leather armour {!*}?
```

Swapping amulets inscribed with `!*` via `e` command prompts four times:

```
Really equip z - an amulet of regeneration {!*}?    <--- use_an_item_menu() calls check_warning_inscriptions() for OPER_EQUIP
                                                        _puton_amulet() is responsible for the next two prompts:
Really put on z - an amulet of regeneration {!*}?   <---    check_warning_inscriptions() for OPER_PUTON
Really remove O - an amulet of reflection {!*}?     <---        check_old_item_warning() for OPER_PUTON
Really remove O - an amulet of reflection {!*}?     <--- remove_ring() calls check_warning_inscriptions() for OPER_REMOVE
```

**Comment by nlavsky** [comment]
Fixed in 91dd1e5a54. Thanks, @rawlins!
